contracts/src/types/ProofRequest.sol

```diff
@@ -61,32 +62,36 @@ library ProofRequestLibrary {
         );
     }
 
-    /// @notice Verifies the client's signature over the proof request.
+    /// @notice Verifies a signature over a proof request.
+    /// @dev Supports EIP-1271 for smart contract signatures. If the signature is empty and the client is a smart
+    /// contract, calls isValidSignature with the abi encoded request as the signature.
```

Comment:
When I first read this comment, I understood it to mean that EIP1271 is used if and only if `signature.len == 0 && addr.code.length > 0`. I then misread the code with this in mind. Now I understand this is simply stating that when the signature is empty, there is a special case to use the request as the "signature".